src/main/java/com/salesforce/storm/spout/dynamic/coordinator/SpoutMonitor.java

```diff
@@ -203,9 +210,6 @@ public void run() {
 
             // Start monitoring loop.
             while (keepRunning()) {
-                // Look for completed tasks
```

Comment:
The logic here got moved to the CompleteableFuture.handle() method below, which allows us to remove keeping a copy of the CompletableFuture (ie spoutThreads class property removed above).  It also means the code that handles when a thread ends will fire at the time the thread ends, instead of having to poll for it.